lib/datadog/statsd/message_buffer.rb

```diff
@@ -42,16 +42,24 @@ def add(message)
         true
       end
 
+      def reset
+        buffer.clear
+        @message_count = 0
+      end
+
+      def reset_telemetry
+        connection.reset_telemetry
+      end
```

Comment:
I'm curious, are we not calling `#reset_telemetry` in `#reset` because we have two use cases for it:
1. `#reset` after a `#flush` (want to keep stats)
2. `#reset` after a fork (want to drop stats)

If that's the case, perhaps we could model this in a different way:

```ruby
def reset
  clear_buffer
  connection.reset_telemetry
end
```

and

```ruby
def initialize
   # ...
   @buffer = String.new
   clear_buffer
end

def flush
   # ...
   connection.write(buffer)

   clear_buffer
end

private

def clear_buffer
  buffer.clear
  @message_count = 0
end
```

This way we don't leak out the "telemetry" concern out of this object to the callers.

Reply:
Yup that was the reason. Works for me as well, I'll use this it's clearer from outside of `MessageBuffer` 👍